base/collection_n1ql.go

```diff
@@ -0,0 +1,125 @@
+package base
+
+import (
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/couchbase/gocb"
+	sgbucket "github.com/couchbase/sg-bucket"
+	pkgerrors "github.com/pkg/errors"
+)
+
+var _ N1QLStore = &Collection{}
+
+// Keyspace for a collection is bucket name until wider collection support is added
+func (c *Collection) Keyspace() string {
+	return c.Bucket().Name()
+}
+
+func (c *Collection) Query(statement string, params map[string]interface{}, consistency ConsistencyMode, adhoc bool) (resultsIterator sgbucket.QueryResultIterator, err error) {
+
+	bucketStatement := strings.Replace(statement, KeyspaceQueryToken, c.Bucket().Name(), -1)
+
+	n1qlOptions := &gocb.QueryOptions{
+		ScanConsistency: gocb.QueryScanConsistency(consistency),
+		Adhoc:           adhoc,
+		NamedParameters: params,
+	}
+
+	waitTime := 10 * time.Millisecond
+	for i := 1; i <= MaxQueryRetries; i++ {
+		Tracef(KeyQuery, "Executing N1QL query: %v", UD(bucketStatement))
```

Comment:
Could log params too, although appreciate this is only trace logging.

```suggestion
		Tracef(KeyQuery, "Executing N1QL query: %v - %v", UD(bucketStatement), UD(params))
```